### PR TITLE
Fix bot.setCommandBlock() for versions 1.13-1.16 and add tests

### PR DESCRIPTION
**Pull request by ImHarvol**
Related to #1325

I dont know why `standard --fix` did such a mess, sorry

### CODE REVIEW

**Comment by rom1504** [comment]
not passing the tests

**Review by wvffle — APPROVED** [review]

File: lib/plugins/command_block.js [hunk 1]
```diff
@@ -1,101 +1,114 @@
...
 const ProtoDef = require('protodef').ProtoDef
 
 module.exports = inject
 
 function inject (bot) {
-  function setCommandBlock (pos, command, trackOutput) {
-    const pluginChannelName = bot.supportFeature('usesAdvCdm') ? 'MC|AdvCdm' : 'MC|AdvCmd'
-
-    const proto = new ProtoDef()
+  function setCommandBlock (pos, command, trackOutput, flags) {
```
Comment (rom1504) [inline lib/plugins/command_block.js:8]: Can you update the doc with this flags parameter please ?

**Review by rom1504 — CHANGES_REQUESTED** [review]

File: lib/plugins/command_block.js (hunk 1 above)
Comment (ImHarvol) [reply lib/plugins/command_block.js:8]: The thing is that I'm not sure which parameters it accepts and what they do
https://wiki.vg/Protocol#Update_Command_Block

Comment (rom1504) [reply lib/plugins/command_block.js]: ah yes sorry I forgot this pr
so basically if you don't know what it does, probably you should expose it in the API
mineflayer should abstract away as much as possible the protocol details, so either remove that flags from the exposed API, either clarify in the doc what values it take and make sure that's stable with versions

**Comment by rom1504** [comment]
left a comment above about mode what is it ?

**Review by rom1504 — APPROVED** [review]

**Comment by rom1504** [comment]
ok let's merge for now, but no guaranty of stability on that mode param, might go at any time

**Pull request merged**